crates/m3/src/builder/witness.rs

```diff
@@ -154,6 +154,50 @@ impl<'cs, 'alloc, F: TowerField, P: PackedField<Scalar = F>> WitnessIndex<'cs, '
 			.collect()
 	}
 
+	pub fn repack<'cs_2, P2: TryRepackSliceInplace<P, Scalar: TowerField>>(
+		self,
+		converted_cs: &'cs_2 ConstraintSystem<P2::Scalar>,
+	) -> Result<WitnessIndex<'cs_2, 'alloc, P2>, super::Error> {
+		let mut tables = Vec::with_capacity(self.tables.len());
+		for table in self.tables {
+			let converted_table = match table {
+				Either::Right(table) => {
+					let cols = table
+						.cols
+						.into_iter()
+						.map(|col| {
+							let data = col.data.try_repack::<P2>()?;
+							Ok(WitnessIndexColumn::<'_, P2> {
+								shape: col.shape,
+								data,
+								is_single_row: col.is_single_row,
+							})
+						})
+						.collect::<Result<Vec<_>, super::Error>>()?;
+					let table_id = table.table.id;
+					Either::Right(TableWitnessIndex {
+						table: converted_cs.get_table(table_id),
+						oracle_offset: table.oracle_offset,
+						cols,
+						size: table.size,
+						log_capacity: table.log_capacity,
+						min_log_segment_size: table.min_log_segment_size,
+					})
+				}
+				Either::Left(table) => {
+					let table_id = table.id;
+					Either::Left(converted_cs.get_table(table_id))
+				}
+			};
+
+			tables.push(converted_table);
+		}
+		Ok(WitnessIndex {
+			tables,
+			allocator: self.allocator,
+		})
+	}
+
 	pub fn into_multilinear_extension_index(self) -> MultilinearExtensionIndex<'alloc, P>
 	where
 		P: PackedExtension<B1>
```

Comment:
Please apply yesterday's changes of `into_multilinear_extension_index`, using <PackedTower> instead of the PackedExtension's of P